metis-common/metis-common-zoho/src/main/java/eu/europeana/metis/zoho/ZohoAccessClient.java

```diff
@@ -247,31 +248,40 @@ public ZohoAccessClient(TokenStore tokenStore, String zohoEmail, String clientId
           new IllegalArgumentException(
               String.format("Provided page: %s, and pageSize: %s", page, pageSize)));
     }
-    int start = ((page - 1) * pageSize) + 1;
+//    int start = ((page - 1) * pageSize) + 1;
 
     try {
       APIResponse<ResponseHandler> response;
       RecordOperations recordOperations = new RecordOperations();
       ParameterMap paramInstance = new ParameterMap();
       if (isNullOrEmpty(searchCriteria)) {//No searchCriteria available
-        paramInstance.add(GetRecordsParam.PAGE, start);
+        paramInstance.add(GetRecordsParam.PAGE, page);
         paramInstance.add(GetRecordsParam.PER_PAGE, pageSize);
         HeaderMap headerInstance = new HeaderMap();
         headerInstance.add(GetRecordsHeader.IF_MODIFIED_SINCE, modifiedDate);
         response = recordOperations
             .getRecords(ZohoConstants.ACCOUNTS_MODULE_NAME, paramInstance, headerInstance);
 
       } else {
-        paramInstance.add(SearchRecordsParam.PAGE, start);
+        paramInstance.add(SearchRecordsParam.PAGE, page);
         paramInstance.add(SearchRecordsParam.PER_PAGE, pageSize);
         paramInstance.add(SearchRecordsParam.CRITERIA,
             createZohoCriteriaString(searchCriteria, criteriaOperator));
+        
         response = recordOperations
             .searchRecords(ZohoConstants.ACCOUNTS_MODULE_NAME, paramInstance);
       }
-      return getZohoRecords(response);
+      
+      if(response.getStatusCode() == 200) {
```

Comment:
Is this if else block needed?
Doesn't the getZohoRecords method cover the 204 code and all other codes with 
eu/europeana/metis/zoho/ZohoAccessClient.java:141?